### PR TITLE
451006 - YauheniiKhudzik

### PR DESCRIPTION
**Pull request by MartinEmbers**
Finished lab, and would appreciate some code review.

### CODE REVIEW

File: Tracer/MethodInfo.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        {
+            firstLevelChildren.Add(methodInfo);
+        }
+
+        internal void StopMethodTrace()
+        {
+            stopWatch.Stop();
+        }
+
+        internal bool isEquals(MethodInfo methodInfo)
```
Comment (RamanBut-Husaim) [inline Tracer/MethodInfo.cs]: @MartinEmbers `PascalCase` for method names.

File: Tracer/PlainTextTraceResultFormatter.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    {
+        private readonly Stream stream;
+        public PlainTextTraceResultFormatter(Stream stream)
+        {
+            this.stream = stream;
+        }
+
+        public void Format(TraceResult traceResult)
+        {
+            if (traceResult == null) return;
```
Comment (RamanBut-Husaim) [inline Tracer/PlainTextTraceResultFormatter.cs]: @MartinEmbers Seems that a better approach is to throw `ArgumentNullException` as this is an invalid input for the class.

File: Tracer/XmlTraceResultFormatter.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    {
+        private readonly Stream stream;
+        public XmlTraceResultFormatter(Stream stream)
+        {
+            this.stream = stream;
+        }
+
+        public void Format(TraceResult traceResult)
+        {
+            if (traceResult == null) return;
```
Comment (RamanBut-Husaim) [inline Tracer/XmlTraceResultFormatter.cs]: @MartinEmbers The same issue as for ConsoleFormatter

**Comment by MartinEmbers** [comment]
The issues have been fixed.

**Comment by MartinEmbers** [comment]
The issues have been fixed.

**Comment by MartinEmbers** [comment]
According to your recommendation about TraceResult structure, I have added some code to the project. Can you check it please?:)

**Comment by MartinEmbers** [comment]
Ready to merge

**Comment by RamanBut-Husaim** [comment]
@MartinEmbers Accepted. Thank you.

**Pull request merged**